lib/faexport/scraper.rb

```diff
@@ -489,7 +489,228 @@ def submit_journal(title, description)
+                watch_id: "",
+                name: "Removed by the user",
+                profile: fa_address,
+                profile_name: "",
+                avatar: fa_url(elem.at_css("img")['src']),
+                posted: "",
+                posted_at: ""
+            }
+          end
+          next
+        end
+        date = pick_date(elem.at_css('.popup_date'))
+        new_watches << {
+            watch_id: elem.at_css("input")['value'],
+            name: elem.at_css("span").content,
+            profile: fa_url(elem.at_css("a")['href']),
+            profile_name: last_path(elem.at_css("a")['href']),
+            avatar: "https:#{elem.at_css("img")['src']}",
+            posted: date,
+            posted_at: to_iso8601(date)
+        }
+      end
+    end
+    # Parse new submission comments notifications
+    new_submission_comments = []
+    if html.at_css("fieldset#messages-comments-submission")
+      html.at_css("fieldset#messages-comments-submission").css("li:not(.section-controls)").each do |elem|
+        if elem.at_css("input")['checked'] == "checked"
+          if include_deleted
+            new_submission_comments << {
+                comment_id: "",
+                name: "Comment or the submission it was left on has been deleted",
+                profile: fa_address,
+                profile_name: "",
+                is_reply: nil,
+                your_submission: nil,
+                submission_id: "",
+                title: "Comment or the submission it was left on has been deleted",
+                posted: "",
+                posted_at: ""
+            }
+          end
+          next
+        end
+        elem_links = elem.css("a")
+        date = pick_date(elem.at_css('.popup_date'))
+        new_submission_comments << {
+            comment_id: elem.at_css("input")['value'],
+            name: elem_links[0].content,
+            profile: fa_url(elem_links[0]['href']),
+            profile_name: last_path(elem_links[0]['href']),
+            is_reply: elem.to_s.include?("<em>your</em> comment on"),
+            your_submission: elem.css('em').last.content == "your",
+            submission_id: elem_links[1]['href'].split("/")[-2],
+            title: elem_links[1].content,
+            posted: date,
+            posted_at: to_iso8601(date)
+        }
+      end
+    end
+    # Parse new journal comments notifications
+    new_journal_comments = []
+    if html.at_css("fieldset#messages-comments-journals")
+      html.at_css("fieldset#messages-comments-journals").css("li:not(.section-controls)").each do |elem|
+        if elem.at_css("input")['checked'] == "checked"
+          if include_deleted
+            new_journal_comments << {
+                comment_id: "",
+                name: "Comment or the journal it was left on has been deleted",
+                profile: fa_address,
+                profile_name: "",
+                is_reply: nil,
+                your_journal: nil,
```

Comment:
I would default to false for these two.  Any code expecting to parse a `bool` will get thrown off by a `null` value.

Reply:
Alright, switched that in 2d39c9b774f2c87cc5bd8db14ea95d83820d7d16